bee-protocol/src/worker/bundle_validator.rs

```diff
@@ -42,14 +43,15 @@ impl<N: Node> Worker<N> for BundleValidatorWorker {
             while let Some(BundleValidatorWorkerEvent(hash)) = receiver.next().await {
                 match load_bundle_builder(tangle(), &hash) {
                     Some(builder) => {
-                        if builder.validate().is_ok() {
+                        if let Ok(bundle) = builder.validate() {
                             tangle().update_metadata(&hash, |metadata| {
                                 metadata.flags_mut().set_valid(true);
-                            })
+                            });
+                            tangle().insert_tip(hash, *bundle.trunk(), *bundle.branch());
```

Comment:
Just double checking are we sure that's what we want ? These are **NOT** the trunk/branch of the tip but of the bundle.

Reply:
Yes! Once the bundle is validated we can put it into the pool.
The transaction is solid and has OTRSI/YTRSI values set.
Or to what are you referring exactly?

Reply:
The trunk/branch are the referenced tails of other bundles right? @thibault-martinez